### PR TITLE
Feature packet queue

### PR DESCRIPTION
**Pull request by Lansenou**

### CODE REVIEW

File: Confus/ConfusServer/Audio/PlayerAudioEmitter.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            createAudioSources();
+        }
+
+        PlayerAudioEmitter::~PlayerAudioEmitter()
+        {
+            delete(m_AudioSourceFootstep1);
+            delete(m_AudioSourceFootstep2);
+            delete(m_AudioSourceFootstep3);
+            delete(m_AttachedPlayer);
```
Comment (beerhond) [inline Confus/ConfusServer/Audio/PlayerAudioEmitter.cpp:17]: My error is still in here.. This probably won't get updated with develop right?

Comment (Lansenou) [inline Confus/ConfusServer/Audio/PlayerAudioEmitter.cpp:17]: We're going to need a system to keep the ConfusServer project updated with the Confus project. @Athosvk, do you have an idea for this?

File: Confus/ConfusServer/Flag.cpp [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+		}
+	}
+
+	//TODO Score points to team of a_PlayerObject
+	void Flag::score(Player* a_PlayerObject) 
+    {
+        *a_PlayerObject->CarryingFlag = EFlagEnum::None;
+	}
+
+	//TODO Drop Flag at position of player
```
Comment (ChristianStiehl) [inline Confus/ConfusServer/Flag.cpp]: not sure if these TODO comments should be here.

File: Confus/ConfusServer/Game.cpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        m_CurrentTicks = m_Device->getTimer()->getTime();
+        m_DeltaTime = (m_CurrentTicks - m_PreviousTicks) / 1000.0;
+
+        m_PlayerNode.update();
+        m_Listener.setPosition(m_PlayerNode.CameraNode->getAbsolutePosition());
+        irr::core::quaternion playerRotation(m_PlayerNode.CameraNode->getRotation());
+
+        //Todo: Fix rotations
+        irr::core::vector3df upVector = playerRotation * irr::core::vector3df( 0, 1, 0 );
+        irr::core::vector3df forwardVector = playerRotation * irr::core::vector3df(0, 0, 1);
```
Comment (ChristianStiehl) [inline Confus/ConfusServer/Game.cpp]: does this work? Otherwise it should maybe be its own task

File: Confus/ConfusServer/Game.cpp [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+			m_MazeGenerator.refillMainMaze(static_cast<int>(time(0)));
+		}
+		m_MazeGenerator.fixedUpdate();
+    }
+
+    void Game::render()
+    {
+        m_Device->getVideoDriver()->beginScene(true, true, irr::video::SColor(255, 100, 101, 140));
+        m_Device->getSceneManager()->drawAll();
+        //m_Device->getGUIEnvironment()->drawAll();
```
Comment (ChristianStiehl) [inline Confus/ConfusServer/Game.cpp:163]: line should be removed, not commented?

**Comment by beerhond** [comment]
Looks good to me. Didn't actually test this.

**Comment by Lansenou** [comment]
@ChristianStiehl  I removed the drop flag todo's, but the score todo's need to happen as soon as we can add a score/points.

**Pull request merged**